thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -253,7 +255,23 @@ public Builder registerAction(
                 @NonNull String alias,
                 @NonNull String actionName,
                 @NonNull Class<? extends Action> actionClass){
-            //TODO: validate actionClass, should contain only 1 field
+            // validate field of actionClass
+            List<Field> fields = new ArrayList<>();
+            for (Field field : actionClass.getDeclaredFields()) {
+                if (!field.getName().equals("this$0")){ // in case of inner class, has this field
```

Comment:
@yongpingchen Can we create instances non static inner class with Gson?

Creating instances of non static inner class requires an instance of parent class. I think we can not provide the instance of parent class.

What do you think?